documentation/nat-service/nat-service.md

```diff
@@ -0,0 +1,225 @@
+
+* A routing lookup is performed based on destination VTEP IP address before sending the VXLAN encapsulated packet to VM.
+
+### 2.2 Outbound packet processing pipeline
+
+![dnat-outbound-packet-processing-pipeline](images/Figure5-dnat-outbound-pipeline.png)
+*Figure 5 - DNAT outbound packet processing pipeline*
+
+* As described in section 2.1, the outbound flow table entry has been created during the handling of the *first packet* processing, when the responding packet arrives from VM, it matches the entry and performs the specified action of replacing the source IP address/port number and removing the VXLAN header before being sent to the internet.
+
+* Then it routes the packet based on DIP of the decapsulated packet.
+
+* Counting and/or metering is done, and they are optional based on the VNET user's billing mode.
+
+● The packet is sent to Internet via the selected interface.
+
+## Configuration example
+
+``` JSON
+
+/* Define Vnet1 */
+DASH_VNET:Vnet1 {
+    "vni": 12345
+}
+
+/* Define Vnet2 */
+DASH_VNET:Vnet2 {
+    "vni": 45678
+}
+
+/* Define routing types */
+DASH_ROUTING_TYPE:vnet_encap: [
+    {
+         "name": "action1",
+         "action_type: "staticvxlan",
+         "encap_type" "vxlan"
+    }
+]
+
+/* Define Overlay routing tables */
+DASH_VNET_ROUTE_TABLE:Vnet1:10.1.0.10 {
+    "routing_type":"vnet_encap", 
+    "dst_vtep":[30.0.0.22]
+    "mac_address":002244AABBCC
+},
+DASH_VNET_ROUTE_TABLE:Vnet1:10.1.0.2 {
+    "routing_type":"vnet_encap", 
+    "dst_vtep":[30.0.0.23]
+    "mac_address":002244AABBDD
+}
+
+
+/* Define SNAT rules */
+DASH_SNAT_RULE_TABLE:Vnet1:10.1.0.0/24 {
+    "0.0.0.0/0": {
+        "src_ip_list":[111.1.1.237, 111.1.1.238]
+    }
+},
+DASH_SNAT_RULE_TABLE:Vnet1:10.1.1.0/24 {
+    "0.0.0.0/0": {
+        "src_ip_list":[111.1.1.239, 111.1.1.240]
+    }
+},
+DASH_SNAT_RULE_TABLE:Vnet2:10.1.0.0/24 {
+    "0.0.0.0/0": {
+        "src_ip_list":[111.1.1.241]
+    },
+    "102.0.0.0/8": {
+        "src_ip_list":[111.1.1.242]
+    },
+}
+
+/* Define DNAT rules */
+DASH_DNAT_RULE_TABLE:111.2.190.195:80 {
+    "mapping_vnet": Vnet1,
+    "mapping_address": 10.1.0.2:80
+},
+DASH_DNAT_RULE_TABLE:114.66.253.4:8080 {
+    "mapping_vnet": Vnet1,
+    "mapping_address": 10.1.0.2:80
+},
+DASH_DNAT_RULE_TABLE:111.2.190.196 {
+    "mapping_vnet": Vnet1,
+    "mapping_address": 10.1.0.3
+},
+DASH_DNAT_RULE_TABLE:114.66.253.5 {
+    "mapping_vnet": Vnet1,
+    "mapping_address": 10.1.0.3
+},
+
+/* Define NAT gateway vtep ip */
+DASH_NAT_GW_VTEP_TABLE{
+    "vtep_ip": [30.0.0.250]
+}
+
+```
+
+## Packet flow example
+
+### SNAT packet flow
+
+`Source: VM in Vnet1, VNI 12345, SIP 10.1.0.10`
+`Destination: IP 8.8.8.8, port 53, protocol UDP`
+
+* **first packet** processing flow of outbound packet
+
+![snat-first-packet-processing-outbnand-packet-flow](images/Figure6-snat-first-packet-processing-outbound-packet-flow.png)
+*Figure 6 - SNAT first packet processing outbound packet flow*
```

Comment:
Where does it get the sport=45678 to rewrite in the packet?  This sport is not part of the SNAT_RULE_TABLE so not sure where it comes from?